### ISSUE TITLE
Support input validation in quick pick

### ISSUE THREAD

**Issue by sandy081** [microsoft/vscode#187863]
Support input validation in quick pick

<img width="629" alt="image" src="https://github.com/microsoft/vscode/assets/10746682/d700f931-b8cf-4903-ab31-1685d02c48f8">

or

<img width="627" alt="image" src="https://github.com/microsoft/vscode/assets/10746682/55a3c07d-d3fd-424e-9ea0-d63425ffe52c">

CC @daviddossett

**Comment by TylerLeonhardt** [comment]
This PR implements the first. I approved it so that @sandy081 is unblocked 
https://github.com/microsoft/vscode/pull/187862
but I have a preference to the 2nd option because it's more consistent with the rest of the product like:
<img width="443" alt="image" src="https://github.com/microsoft/vscode/assets/2644648/bad75beb-c4fd-4ae3-8241-19556982ab02">

**Comment by sandy081** [comment]
It seems floating breaks if Severity is Ignore:

<img width="643" alt="image" src="https://github.com/microsoft/vscode/assets/10746682/03848b5a-d886-4859-be32-42342f44b83a">

**Comment by daviddossett** [comment]
+1 for rendering as a popover vs. pushing the quick pick content down.

**Comment by sandy081** [comment]
Wondering how do you represent the message when Severity is Ignore, like in my previous comment?

**Comment by TylerLeonhardt** [comment]
I think `Severity.Ignore` probably doesn't show anything. It's "Ignored" so we shouldn't show anything.

**Comment by alexr00** [comment]
+1 for this and for adding extension API for it!

**Comment by lppedd** [comment]
Would be nice to have this. We want to both allow the user to input an arbitrary value, and possibly select proposed items.
Without validation the UX is pretty bad and users tend to not realize they're inputting something incorrect. Thus they have to start over multiple times.
